src/lib/config-path.js

```diff
@@ -1,110 +1,114 @@
+
+import {folder_exists, file_exists} from './utils';
 
-const fs = require('fs');
-const Path = require('path');
-const logger = require('./logger');
 const CONFIG_FILE = 'config.yaml';
+const XDG = 'xdg';
+const WIN = 'win';
+const WIN32 = 'win32';
 const pkgJson = require('../../package.json');
+
 /**
  * Find and get the first config file that match.
  * @return {String} the config file path
  */
-function find_config_file() {
-  const paths = get_paths();
+function findConfigFile(configPath) {
+  if (_.isNil(configPath) === false) {
+    return Path.resolve(configPath);
+  }
 
-  for (let i=0; i<paths.length; i++) {
-    if (file_exists(paths[i].path)) return paths[i].path;
+  const configPaths = getConfigPaths();
+
+  if (_.isEmpty(configPaths)) {
+    throw new Error('no configuration files can be proccesed');
   }
 
-  create_config_file(paths[0]);
-  return paths[0].path;
+  const primaryConf = _.find(configPaths, (configLocation) => file_exists(configLocation.path));
+  if (_.isNil(primaryConf) === false) {
+    return primaryConf.path;
+  }
+
+  return createConfigFile(_.head(configPaths)).path;
 }
 
-/**
- * Create a default config file in your system.
- * @param {String} config_path
- */
-function create_config_file(config_path) {
-  require('mkdirp').sync(Path.dirname(config_path.path));
-  logger.logger.info({file: config_path.path}, 'Creating default config file in @{file}');
+function createConfigFile(configLocation) {
+  createConfigFolder(configLocation);
+
+  const defaultConfig = updateStorageLinks(configLocation, readDefaultConfig());
 
-  let created_config = fs.readFileSync(require.resolve('../../conf/default.yaml'), 'utf8');
+  fs.writeFileSync(configLocation.path, defaultConfig);
+
+  return configLocation;
+}
+
+function readDefaultConfig() {
+  return fs.readFileSync(require.resolve('../../conf/default.yaml'), 'utf8');
+}
+
+function createConfigFolder(configLocation) {
+  mkdirp.sync(Path.dirname(configLocation.path));
+  logger.logger.info({file: configLocation.path}, 'Creating default config file in @{file}');
+}
+
+function updateStorageLinks(configLocation, defaultConfig) {
+    if (configLocation.type !== XDG) {
+        return defaultConfig;
+    }
 
-  if (config_path.type === 'xdg') {
     // $XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored,
     // If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.
-    let data_dir = process.env.XDG_DATA_HOME|| Path.join(process.env.HOME, '.local', 'share');
-    if (folder_exists(data_dir)) {
-      data_dir = Path.resolve(Path.join(data_dir, pkgJson.name, 'storage'));
-      created_config = created_config.replace(/^storage: .\/storage$/m, `storage: ${data_dir}`);
+    let dataDir = process.env.XDG_DATA_HOME || Path.join(process.env.HOME, '.local', 'share');
+    if (folder_exists(dataDir)) {
+      dataDir = Path.resolve(Path.join(dataDir, pkgJson.name, 'storage'));
```

Comment:
JSON <- All uppercase in http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf
Line 77: getXDGHome

Maybe we should change `Json` to `JSON` or change `XGD` to `Xdg`